### DISCUSSION TITLE
Blazor Hybrid support through AvaloniaUI

### DISCUSSION THREAD

**Discussion by JensNordenbro** [AvaloniaUI/Avalonia#16206] [Ideas]
Today Blazor Hybrid is a model to develop hybrid desktop applications.
Linux is left out of the equation.
Please make AvaloniaUI the goto solution for Blazor Hybrid on Linux by for example expanding on this effort https://github.com/jpmikkers/Baksteen.Avalonia.Blazor

**Comment by JensNordenbro** [comment]
This would drive adoption of AvaloniaUI as a nice side effect.

**Comment by aabluedragon** [comment]
Been looking for the exact same use case, I really hope it ends up in AvaloniaUI v11.
Blazor Hybrid apps are becoming more common, for those looking to create cross platform web apps with native capabilities.
Tested out https://github.com/jpmikkers/Baksteen.Avalonia.Blazor and it worked nicely, I really like to see something like that officially supported by AvaloniaUI, as a programming path.

**Comment by Kukks** [comment]
I would also very much like this feature. Maui Linux support is non-existent and unplanned and is key for Blazor cross-platform applications.

**Comment by davidchavezgrant** [comment]
Would really love to have this, considering the state of MAUI currently.

**Comment by scottkuhl** [comment]
This is perfect for filling in the gaps in Avalonia, like using the SyncFusion Blazor PDF Viewer but having the majority of the app in Avalonia.

**Comment by Heartnett** [comment]
AvaloniaUI should have first-class support for Blazor.

**Comment by rwecho** [comment]
Blazor Hybrid is a excellent architecture, consider it.

**Comment by maxkatz6** [comment]
> Please make AvaloniaUI the goto solution for Blazor Hybrid on Linux by for example expanding on this effort

Lack of a standard native web view control on Linux is a problem. Alternatively, you would have to rely on user pre-installed gtk with webview, or ship your app with chromium.

**Reply by scottkuhl** [reply]
Uno Platform seems to be running into the same issue.  They have chosen to move forward but not support Linux yet.  **Is there an option to move forward with Blazor Hybrid but not support Linux?**  Or is it against Avalonia principle to not officially support something unless it reaches all desktop platforms?  I think that still solves problems for many people, if not all.

> WebView2 is currently supported on Windows, Android, iOS, and macOS.

https://platform.uno/docs/articles/controls/WebView.html

**Reply by maxkatz6** [reply]
> Or is it against Avalonia principle to not officially support something unless it reaches all desktop platforms?

We don't have such principle, though it's really ideal if we could support it everywhere.
I would say instead, that blazor hybrid support is not on our plans in general. We are not against it as a community project though.

**Reply by martinbu** [reply]
I think it would not be an issue if gtk with webview is a requirement for linux users. For Windows, WebView2 must also be installed - which is not the case in every setup (Enterprises can disable it). Also Photino.NET uses this approach. Maybe the Avalonia project could integrate the needed parts from the photino.NET project?
https://github.com/tryphotino/photino.NET
https://github.com/tryphotino/photino.Blazor

It would be great if Avalonia would support a WebView2 out of the box. Even better with Balzor Hybrid support. For us it is the main blocker for not using Avalonia.

There are other projects that try to achive this...
https://github.com/BeyondDimension/Avalonia.WebView2

**Reply by vlad-lubenskyi** [reply]
@martinbu there are CEF-based web views for Avalonia and it relatively easy to integrate it all with Blazor Hybrid. We did this for DotNetBrowser, which is commercial, but also wrote an [article](https://teamdev.com/dotnetbrowser/blog/blazor-hybrid-apps-in-avalonia-ui/) that explains how this integration works. It shouldn't be to complicated to do the same with CEF.

**Comment by bxjg1987** [comment]
I really, really like this feature
